R/url_path.R

```diff
@@ -0,0 +1,20 @@
+#' Construct URL Path
+#' 
+#' Construct a URL from path components like \code{file.path}
+#'
+#' @param x a URL, or vector of URLs
+#' @param ... character vectors
+#' 
+#' @examples
+#' 
+#' url_path("https://github.com", c('Ironholds', 'nfultz'), 'urltools')
```

Comment:
Length 2 vector containing:
"https://github.com/Ironholds/urltools"  "https://github.com/nfultz/urltools"

same as file.path would.

For log files that have a format like s3://bucket/YYYY/MM/DD/name.log it's nice to be able to vectorize this operation.